### ISSUE TITLE
Duplicate method Error on @QuarkusTest when PanacheRepositoryBase method is @Overrided

### ISSUE THREAD

**Issue by lssoares** [quarkusio/quarkus#6374]
**Describe the bug**
When a PanacheRepositoryBase is @Overrided, tests on a @QuarkusTests annotated class will fail with a duplicated method message error.

**Expected behavior**
Tests should run using the method code that overrides the PanacheRepositoryBase method.

**Actual behavior**
Tests fail with the following error:
```
[ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 3.816 s <<< FAILURE! - in org.acme.ExampleResourceTest
[ERROR] testHelloEndpoint  Time elapsed: 0.01 s  <<< ERROR!
org.junit.jupiter.api.extension.TestInstantiationException: TestInstanceFactory [io.quarkus.test.junit.QuarkusTestExtension] failed to instantiate test class [org.acme.ExampleResourceTest]
Caused by: java.lang.ExceptionInInitializerError
Caused by: java.lang.RuntimeException: Failed to start quarkus
Caused by: java.lang.RuntimeException: Failed to initialize Arc
Caused by: java.lang.ClassFormatError: Duplicate method name "listAll" with signature "()Ljava.util.List;" in class file org/acme/ExampleRepository
```

**To Reproduce**
1. Create a class and method like the following:
```
@ApplicationScoped
public class ExampleRepository implements PanacheRepository<ExampleEntity> {

    @Override
    public List<ExampleEntity> listAll() {
          List<ExampleEntity> a = ExampleEntity.<ExampleEntity> findAll().list();
          return a;
    }

}
```

2. Create a Test class and method like the following:
```
@QuarkusTest
public class ExampleResourceTest {

    @Test
    public void getList() {
        List<?> list = CDI.current().select(ExampleRepository.class).get().listAll();
        Assertions.assertTrue(Objects.nonNull(list) && list.size() > 0);
    }
}
```

3. Run the tests

**Configuration**
I guess the problem will happen even with no properties supplied. For the sake of compliance, I'm pasting all the properties used on the project that was created for you to check the problem.

```properties
# Logging
quarkus.log.console.enable=true
quarkus.log.console.format=%d{yyyy-MM-dd HH:mm:ss} %-5p [%c{3.}]] (%t) %s%e%n
quarkus.log.console.level=DEBUG
quarkus.log.console.color=true

# Database
quarkus.datasource.driver=org.h2.Driver
quarkus.datasource.url=jdbc:h2:tcp://localhost/mem:example
quarkus.datasource.username=sa
quarkus.datasource.password=as
quarkus.datasource.min-size=1
quarkus.datasource.max-size=5


# Flyway minimal config properties
quarkus.flyway.migrate-at-start=false

```

**Screenshots**
Not applicable...

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
```
Darwin MacBook-Pro-xxxxxx.local 18.7.0 Darwin Kernel Version 18.7.0: Sun Dec  1 18:59:03 PST 2019; root:xnu-4903.278.19~1/RELEASE_X86_64 x86_64
```
 - Output of `java -version`: 
```
openjdk version "11.0.2" 2019-01-15
OpenJDK Runtime Environment 18.9 (build 11.0.2+9)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.2+9, mixed mode)
```

 - GraalVM version (if different from Java): `Not applicable``

 - Quarkus version or git rev: `1.1.0.Final`

**Additional context**
You can find a sample to reproduce the problem on:
https://github.com/lssoares/single-maven-quarkus/tree/DuplicatedMethodOnTestsDueToPanacheMethodOverride

**Comment by irenakezic** [comment]
I think you should inject ExampleRepository in your test and it should work as expected.
```   
@QuarkusTest
public class ExampleResourceTest {

    @Inject
    ExampleRepository exampleRepository;

    @Test
    public void getList() {
    	List<?> list = exampleRepository.listAll();
        Assertions.assertTrue(Objects.nonNull(list) && list.size() > 0);
    }
}
```

**Comment by lssoares** [comment]
Trying to avoid the instance variable as suggested on issue: https://github.com/quarkusio/quarkus/issues/1724

Anyway, the example is just overriding an interface method... it should work either by using @Inject or by manually getting an Instance of the bean. What am I doing wrong?

**Comment by seseso** [comment]
I have the same problem even on a QuarkusTest that doesn't use the Panache Repository. To be clear:
1 - I have in my project a Panache Repository class where I override the method findByIdOptional.
2 - Any "dummy" QuarkusTest that does not use this repository bean will fail with the exceptions:
Caused by: java.lang.ExceptionInInitializerError
Caused by: java.lang.RuntimeException: Failed to start quarkus
Caused by: java.lang.RuntimeException: Failed to initialize Arc

**Comment by loicmathieu** [comment]
@lssoares this has been fixed in 1.3 I think.
I tested your reproducer with 1.4.1.Final and it works so I close the issue.
